awx/main/dispatch/worker/callback.py

```diff
@@ -123,17 +123,9 @@ def flush(self, force=False):
                 except Exception as exc:
                     # if an exception occurs, we should re-attempt to save the
                     # events one-by-one, because something in the list is
-                    # broken/stale (e.g., an IntegrityError on a specific event)
+                    # broken/stale
                     for e in events:
                         try:
-                            if (
-                                isinstance(exc, IntegrityError) and
-                                getattr(e, 'host_id', '')
```

Comment:
This foreign key constraint just doesn't exist anymore at all (go ahead and go check 😄 `\d main_jobevent`).